src/m_menu.c

```diff
@@ -797,12 +800,72 @@ menu_t LoadDef =
   &MainDef,
   LoadMenu,
   M_DrawLoad,
-  80,34, //jff 3/15/98 move menu up
+  M_X_LOADSAVE,34, //jff 3/15/98 move menu up
   0
 };
 
 #define LOADGRAPHIC_Y 8
 
+// [FG] draw a snapshot of the n'th savegame into a separate window,
+//      or fill window with solid color and "n/a" if snapshot not available
+
+static int snapshot_width, snapshot_height;
+
+static void M_DrawBorderedSnapshot (int n)
+{
+  int x, y;
+  patch_t *patch;
+
+  const int snapshot_x = MAX((WIDESCREENDELTA + SaveDef.x + SKULLXOFF - snapshot_width) / 2, 8);
+  const int snapshot_y = LoadDef.y + MAX((load_end * LINEHEIGHT - snapshot_height) * n / load_end, 0);
+
+  // [FG] a snapshot window smaller than 80*48 px is considered too small
+  if (snapshot_width < ORIGWIDTH/4)
+    return;
+
+  if (!M_DrawSnapshot(n, snapshot_x, snapshot_y, snapshot_width, snapshot_height))
+  {
+    char *na = "n/a";
```

Comment:
After this is merged, I'd like to work on const-correctness for `M_WriteText()`, `M_StringWidth()`, `M_StringHeight()` and the like which still get passed non-const `char *` as arguments.